check/checksum.go

```diff
@@ -0,0 +1,178 @@
+package check
+
+import (
+	"bytes"
+	"crypto/md5"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/schollz/progressbar/v3"
+)
+
+func checksumFor(path string) ([]byte, error) {
+	h := md5.New()
+
+	r, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("error opening %s: %w", path, err)
+	}
+	defer r.Close()
+
+	if _, err := io.Copy(h, r); err != nil {
+		return nil, fmt.Errorf("error reading %s content: %w", path, err)
+	}
+
+	return h.Sum(nil), nil
+}
+
+func checkChecksum(src, target string) (bool, error) {
+	name := filepath.Base(src)
+	sc, err := checksumFor(src)
+	if err != nil {
+		return false, fmt.Errorf("error getting %s checksum: %w", src, err)
+	}
+
+	pth := filepath.Join(target, name)
+	tc, err := checksumFor(pth)
+	if err != nil {
+		return false, fmt.Errorf("error getting %s checksum: %w", pth, err)
+	}
+
+	return bytes.Equal(sc, tc), nil
+}
+
+// CheckChecksum compares md5 checksum files from source and target directories.
+func CheckChecksum(src, target string) error {
```

Comment:
Pensando melhor, acho que não faz sentido ter `src` e `target`. O caso de uso é:
- baixei os arquivos do _mirror_
- os MD5 vão estar no mesmo diretório
- quero verificar se os arquivos maiores (mais propensos a darem erro ao baixar) estão íntegros

Ou seja, é tudo no mesmo diretório, não?

Reply:
Faz sentido.

Reply:
Pensei aqui, pode ser útil querer comparar arquivos checksum. Um cenário do tipo:
- Baixei arquivos do servidor da Receita num determinado momento
- Gerei arquivos checksum para esses arquivos
- Baixei arquivos de um mirror que tem arquivos de checksum acompanhados
- Quero comparar os meus arquivos com esse do mirror 

Mas se for um cenário que não faça muito sentido, ou que vá ser muito raro, posso mudar aqui.

Reply:
Boa! Gostei : ) Deixemos então.